### PR TITLE
fix(move-to-finished): consider addition of prioritized jobs when processing last active job

### PR DESCRIPTION
**Pull request by roggervalf**

### CODE REVIEW

File: tests/test_worker.ts [hunk 1]
```diff
@@ -669,14 +669,44 @@ describe('workers', function () {
     const worker = new Worker(queueName, processor, { connection });
     await worker.waitUntilReady();
 
-    // wait for all jobs to enter the queue and then start processing
-    await Promise.all([normalPriority, mediumPriority, highPriority]);
-
     await processing;
```
Comment (roggervalf) [inline tests/test_worker.ts:672]: by awaiting this line is sufficient

File: src/commands/moveToFinished-13.lua [hunk 2]
```diff
@@ -216,10 +216,17 @@ if rcall("EXISTS", jobIdKey) == 1 then -- // Make sure job exists
 
         jobId = rcall("RPOPLPUSH", KEYS[1], KEYS[2])
 
-        -- If jobId is special ID 0:delay, then there is no job to process
         if jobId then
             if string.sub(jobId, 1, 2) == "0:" then
                 rcall("LREM", KEYS[2], 1, jobId)
+
+                -- If jobId is special ID 0:delay (delay grater than 0), then there is no job to process
```
Comment (manast) [inline src/commands/moveToFinished-13.lua:223]: grater -> greater

**Review by manast — APPROVED** [review]
small typo

File: tests/test_worker.ts [hunk 3]
```diff
@@ -669,14 +669,44 @@ describe('workers', function () {
     const worker = new Worker(queueName, processor, { connection });
```
Comment (adiholden) [inline tests/test_worker.ts:669]: @roggervalf I think that 
await Promise.all([...normalPriority,...mediumPriority,...highPriority]);
should be added before creating a new worker, as you need to make sure that all jobs are entered before worker starts processing jobs.
When I run the test with this fixes the test failed as some jobs where added after the worker start processing.

Comment (roggervalf) [reply tests/test_worker.ts:669]: I see, probably in case of dragonfly its needed, ok let me add it back

**Pull request merged**